lib/ansible/modules/cloud/scaleway/scaleway_compute.py

```diff
@@ -0,0 +1,745 @@
+
+    return target_server
+
+
+def restart_server(compute_api, server):
+    print_warning(compute_api, "Starting restart_server")
+    response = compute_api.post(path="servers/%s/action" % server["id"],
+                                data={"action": "reboot"})
+
+    if not response.ok:
+        msg = 'Error during server reboot: (%s) %s' % (response.status_code, response.json)
+        compute_api.module.fail_json(msg=msg)
+
+    wait = compute_api.module.params["wait"]
+
+    if wait:
+        print_warning(compute_api, "(restart_server) Wait is enabled")
+        wait_to_complete_state_transition(compute_api=compute_api, server=server)
+
+    return response
+
+
+def stop_server(compute_api, server):
+    print_warning(compute_api, "Starting stop_server strategy")
+    response = compute_api.post(path="servers/%s/action" % server["id"],
+                                data={"action": "poweroff"})
+
+    print_warning(compute_api=compute_api, msg=response.json)
+    if not response.ok:
+        err_msg = 'Error during server power-off: (%s) %s' % (response.status_code, response.json)
+        compute_api.module.fail_json(msg=err_msg)
+
+    wait = compute_api.module.params["wait"]
+
+    if wait:
+        print_warning(compute_api, "(stop_server) Wait is enabled")
+        wait_to_complete_state_transition(compute_api=compute_api, server=server)
+
+    return response
+
+
+def start_server(compute_api, server):
+    print_warning(compute_api, "Starting start_server strategy")
+    response = compute_api.post(path="servers/%s/action" % server["id"],
+                                data={"action": "poweron"})
+    if not response.ok:
+        msg = 'Error during server poweron: (%s) %s' % (response.status_code, response.json)
+        compute_api.module.fail_json(msg=msg)
+
+    wait = compute_api.module.params["wait"]
+
+    if wait:
+        print_warning(compute_api, "(start_server) Wait is enabled")
+        wait_to_complete_state_transition(compute_api=compute_api, server=server)
+
+    return response
+
+
+def remove_server(compute_api, server):
+    print_warning(compute_api, "Starting remove server strategy")
+    response = compute_api.delete(path="servers/%s" % server["id"])
+    if not response.ok:
+        msg = 'Error during server deletion: (%s) %s' % (response.status_code, response.json)
+        compute_api.module.fail_json(msg=msg)
+
+    wait = compute_api.module.params["wait"]
+
+    if wait:
+        print_warning(compute_api, "(remove_server) Wait is enabled")
+        wait_to_complete_state_transition(compute_api=compute_api, server=server)
+
+    return response
+
+
+def present_strategy(compute_api, wished_server):
+    print_warning(compute_api, "Starting present strategy")
+    changed = False
+    query_results = find(compute_api=compute_api, wished_server=wished_server, per_page=1)
+
+    if not query_results:
+        changed = True
+        if compute_api.module.check_mode:
```

Comment:
Here and below: I would prefer using something like `return changed, ...` than calling ` compute_api.module.exit_json` because the caller [calls `exit_json`](https://github.com/ansible/ansible/pull/38285/files#diff-4259e225b60891dd12355d4205e44926R713) anyway.